spark/src/main/scala/org/apache/spark/sql/delta/DeltaHistoryManager.scala

```diff
@@ -211,6 +211,9 @@ class DeltaHistoryManager(
         start,
         Some(end),
         deltaLog.newDeltaHadoopConf())
+      if (commits.isEmpty) {
+        throw DeltaErrors.noHistoryFound(deltaLog.logPath)
```

Comment:
Is this the right exception to throw here when [start, end) don't cover the entire history? All the other changes look good though.

Reply:
That's what we throw in other such scenarios in the parent function

https://github.com/delta-io/delta/blob/82e940f17f51a0ebeaac0b03441b13875da3c439/spark/src/main/scala/org/apache/spark/sql/delta/DeltaHistoryManager.scala#L313-L316